op-challenger/fault/responder.go

```diff
@@ -0,0 +1,208 @@
+		if err != nil {
+			return nil, err
+		}
+		return NewFaultResponderTxData(txdata), nil
+	}
+}
+
+// Respond takes a [Claim] and executes the response action.
+func (r *faultResponder) Respond(ctx context.Context, response Claim) error {
+	// Build the transaction data.
+	txData, err := r.buildTxData(ctx, response)
+	if err != nil {
+		return err
+	}
+
+	// Send the transaction through the txmgr queue.
+	receiptsCh := make(chan txmgr.TxReceipt[TxData])
+	queue := txmgr.NewQueue[TxData](ctx, r.txMgr, r.MaxPendingTransactions)
+	err = sendTransaction(txData, queue, receiptsCh)
+	if err != nil {
+		return err
+	}
+
+	// Block until the transaction receipt is received.
+	r.handleReceipts(ctx, receiptsCh)
+
+	return nil
+}
+
+// handleReceipts handles the transaction receipts.
+// This is a blocking method.
+func (r *faultResponder) handleReceipts(ctx context.Context, receiptsCh chan txmgr.TxReceipt[TxData]) {
+	for {
+		select {
+		case rec := <-receiptsCh:
+			r.log.Info("responder received receipt", "txHash", rec.Receipt.TxHash, "status", rec.Receipt.Status)
+		case <-r.shutdownCtx.Done():
+			r.log.Info("closing responder receipt handler")
+			return
+		}
+	}
+}
+
+// faultResponderTxData
+type faultResponderTxData struct {
+	data []byte
+	dest *common.Address
+}
+
+// NewFaultResponderTxData returns a new [faultResponderTxData].
+func NewFaultResponderTxData(tx *types.Transaction) *faultResponderTxData {
+	return &faultResponderTxData{
+		data: tx.Data(),
+		dest: tx.To(),
+	}
+}
+
+// Bytes returns the transaction data as a byte slice.
+func (txdata *faultResponderTxData) Bytes() []byte {
+	return txdata.data
+}
+
+// Destination returns the destination address of the transaction.
+func (txdata *faultResponderTxData) Destination() *common.Address {
+	return txdata.dest
+}
+
+// TxData is the interface for the data sent in a transaction.
+type TxData interface {
+	// Bytes returns the transaction data as a byte slice.
+	Bytes() []byte
+	// Destination returns the destination address of the transaction.
+	Destination() *common.Address
+}
+
+// TxQueue is an interface for a transaction queue.
+type TxQueue interface {
+	// Send sends a transaction to the queue.
+	Send(txdata TxData, candidate txmgr.TxCandidate, receiptsCh chan txmgr.TxReceipt[TxData])
+}
+
+// sendTransaction creates & submits a transaction using the provided [TxData].
+// Under the hood, it uses the challenger's [txmgr] to handle transaction sending.
+// Gas estimation is performed using go-ethereum's core IntrinsicGas estimation function.
+// This is a blocking method. It should not be called concurrently.
+func sendTransaction(txdata TxData, queue TxQueue, receiptsCh chan txmgr.TxReceipt[TxData]) error {
+	// Do the gas estimation offline. A value of 0 will cause the [txmgr] to estimate the gas limit.
+	data := txdata.Bytes()
+	intrinsicGas, err := core.IntrinsicGas(data, nil, false, true, true, false)
```

Comment:
We shouldn't be using `IntrinsicGas` but instead do gas estimation. The op-proposer is a better example than the batcher for the challenger.